docs/docs/cli.md

```diff
@@ -165,6 +165,11 @@ If you just want to update a few packages and not all, you can list them as such
 poetry update requests toml
 ```
 
+Note that for this to work, you must not be using an exact version specification in your
+`pyproject.toml` file. In other terms, `poetry update foo` will be a no-op if the `pyproject.toml`
+contains `foo = "2.3"`. In order for foo to be updated to "2.4", you should first edit the
+configuration file to have `foo =  "^2.3"`.
+
```

Comment:
```suggestion
Note that this will not update versions for dependencies outside their version constraints specified
in the `pyproject.toml` file. In other terms, `poetry update foo` will be a no-op if the version constraint 
specified for `toml` is `~2.3` or `2.3` and `2.4` is available. In order for foo to be updated, you must 
update the constraint, for example `^2.3`. You can do this using the `add` command. 

```
A few minor suggestions.